sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystQlSuite.scala

```diff
@@ -49,4 +50,16 @@ class CatalystQlSuite extends PlanTest {
     parser.createPlan("select sum(product + 1) over (partition by (product + (1)) order by 2) " +
       "from windowData")
   }
+
+  test("limit clause: a support in set operation") {
+    val plan1 = parser.createPlan("select key from (select * from t1) x limit 1")
+    assert(plan1.collect{ case w: Limit => w }.size === 1)
+
+    val plan2 = parser.createPlan("select key from (select * from t1 limit 2) x limit 1")
+    assert(plan2.collect{ case w: Limit => w }.size === 2)
+
+    val plan3 = parser.createPlan("select key from ((select * from testData limit 1) " +
+      "union all (select * from testData limit 1)) x limit 1")
+    assert(plan3.collect{ case w: Limit => w }.size === 3)
```

Comment:
can we do this similar to how we compare plans in various optimizer suites, e.g. https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizeInSuite.scala#L84